src/allocationDispatcher.c

```diff
@@ -0,0 +1,55 @@
+#include "allocationDispatcher.h"
+#include <stdlib.h>
+#include <stdio.h>
+#include <assert.h>
+
+ALLOCATION_DISPATCHER* adInitialize(size_t blockSizes[], size_t numBlocks[], size_t numAllocators, void* memoryPool){
+  void* memoryHead = memoryPool;
+  ALLOCATION_DISPATCHER* dispatcher= memoryHead;
+  memoryHead = ((char*) memoryHead)+sizeof(ALLOCATION_DISPATCHER);

```

Comment:
no need to cast to `char *`; `void *` pointer arithmetic should work as expected.